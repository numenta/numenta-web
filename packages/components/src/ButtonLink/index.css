/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.enabled {
  background-color: #1aa0db;
}
.disabled {
  composes: bg-silver from 'tachyons' !important;
}

.primary {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: ph4 from 'tachyons';
  composes: pv3 from 'tachyons';
  composes: white from 'tachyons';
}

.buttonLink {
  composes: br2 from 'tachyons';
  composes: dib from 'tachyons';
  composes: dim from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: link from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: pointer from 'tachyons';
}
