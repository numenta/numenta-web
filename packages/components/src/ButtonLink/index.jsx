// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


/**
 * External link only (browser location) for now.
 */
const ButtonLink = ({children, disabled, onClick, theme, to}) => {
  const classes = [styles.buttonLink, styles[theme]]

  if (disabled) classes.push(styles.disabled)
  else classes.push(styles.enabled)

  return (
    <a
      className={classNames(...classes)}
      href={to}
      onClick={onClick}
    >
      {children}
    </a>
  )
}

ButtonLink.propTypes = {
  children: React.PropTypes.node.isRequired,
  disabled: React.PropTypes.bool,
  onClick: React.PropTypes.func,
  theme: React.PropTypes.string,
  to: React.PropTypes.string.isRequired,
}

ButtonLink.defaultProps = {
  disabled: false,
  theme: 'primary',
}

export default ButtonLink
