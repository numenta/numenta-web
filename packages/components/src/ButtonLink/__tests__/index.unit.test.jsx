import React from 'react'
import renderer from 'react-test-renderer'

import ButtonLink from '../../ButtonLink'


describe('ButtonLink React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <ButtonLink
        onClick={() => {}}
        to="http://numenta.com"
      >
        Title
      </ButtonLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Disabled correctly', () => {
    const component = renderer.create(
      <ButtonLink
        disabled={true}
        onClick={() => {}}
        to="http://numenta.com"
      >
        Title
      </ButtonLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
