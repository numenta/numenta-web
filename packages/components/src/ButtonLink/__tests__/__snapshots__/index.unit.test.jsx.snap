exports[`ButtonLink React component Renders Default correctly 1`] = `
<a
  className="buttonLink primary enabled"
  href="http://numenta.com"
  onClick={[Function]}>
  Title
</a>
`;

exports[`ButtonLink React component Renders Disabled correctly 1`] = `
<a
  className="buttonLink primary disabled"
  href="http://numenta.com"
  onClick={[Function]}>
  Title
</a>
`;
